.github/workflows/build-and-publish.yml

```diff
@@ -48,16 +48,18 @@ jobs:
           echo "Commit hash: $COMMIT_HASH"
           echo "commit_hash=$COMMIT_HASH" >> "$GITHUB_OUTPUT"
 
-  build-and-publish:
-    name: Build and publish
+  check-image-already-published:
+    name: Check whether the image is already published
     runs-on: ubuntu-latest
     needs: get-commit-hash
-    concurrency: build-and-publish-${{ inputs.app_name }}-${{ needs.get-commit-hash.outputs.commit_hash }}
```

Comment:
this concurrency is important to keep. the concurrency statement later on that uses `github.ref` won't work since different github.refs can refer to the same commit hash (e.g. main, origin/main, HEAD, <commit hash>, <version tag>) can all be valid refs that point to the same commit hash, so there'd be a race condition where multiple jobs are trying to build the same commit hash but don't realize it since they are referencing the commit via different refs. that's why we have the separate job beforehand that gets the commit hash.